File: samples/swiftui/Sources/MilanoBridge/MilanoBridge.swift
import MilanoSDK

/// The one doorway between Milano and the design system: generated typed
/// bindings map nodes to component models, renderers register them.
enum MilanoBridge {
    static func registry() -> MilanoRegistry {
        var registry = MilanoRegistry()
        registry.register(ColumnRenderer(), for: "Column")
        registry.register(BannerRenderer(), for: "Banner")
        registry.register(TextRenderer(), for: "Text")
        registry.register(ButtonRenderer(), for: "Button")
        registry.register(TextFieldRenderer(), for: "TextField")
        registry.register(NumberFieldRenderer(), for: "NumberField")
        registry.register(CheckboxRenderer(), for: "Checkbox")
        registry.register(RowRenderer(), for: "Row")
        registry.register(CardRenderer(), for: "Card")
        registry.register(ImageRenderer(), for: "Image")
        return registry
    }
}

File: samples/compose/app/src/main/kotlin/dev/getmilano/sample/milanobridge/MilanoBridge.kt
package dev.getmilano.sample.milanobridge

import dev.getmilano.MilanoRegistry

/**
 * The one doorway between Milano and the design system: generated typed
 * bindings map nodes to component models, renderers register them.
 */
fun milanoRegistry(): MilanoRegistry {
    val registry = MilanoRegistry()
    registry.register("Column", ColumnRenderer)
    registry.register("Banner", BannerRenderer)
    registry.register("Text", TextRenderer)
    registry.register("Button", ButtonRenderer)
    registry.register("TextField", TextFieldRenderer)
    registry.register("NumberField", NumberFieldRenderer)
    registry.register("Checkbox", CheckboxRenderer)
    registry.register("Row", RowRenderer)
    registry.register("Card", CardRenderer)
    registry.register("Image", ImageRenderer)
    return registry
}
